### PR TITLE
fix filecoin bugs

### PR DESCRIPTION
**Pull request by spring820313**
## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] If there is a related Issue, mention it in the description (e.g. 'Fixes #444 ).

### CODE REVIEW

**Comment by hewigovens** [comment]
can you add a test to cover this change?

File: src/Filecoin/Transaction.cpp [hunk 1]
```diff
@@ -12,6 +12,10 @@ using namespace TW::Filecoin;
 // encodeVaruint encodes a 256-bit number into a big endian encoding, omitting leading zeros.
 static Data encodeVaruint(const uint256_t& value) {
     Data data;
+	if(value == uint256_t(0)) {
+		return data;
+	}
+
```
Comment (hewigovens) [inline src/Filecoin/Transaction.cpp:18]:
```suggestion
    if (value == uint256_t(0)) {
        return data;
    }
```

**Comment by hewigovens** [comment]
you need to write tests to fix codecov warning

**Comment by catenocrypt** [comment]
So 0 value is not added to encoded data?  Why?  How can decode return the original value then?  I have doubts.

**Comment by spring820313** [comment]
> So 0 value is not added to encoded data? Why? How can decode return the original value then? I have doubts.

https://github.com/filecoin-project/cpp-filecoin/blob/master/core/primitives/big_int.hpp:18

**Comment by catenocrypt** [comment]
For test: take over from branch ar/filecoinfix .

**Comment by catenocrypt** [comment]
Please fix test coverage, otherwise PR will be closed.  You can take over tests from branch ar/filecoinfix .

**Comment by spring820313** [comment]
> Please fix test coverage, otherwise PR will be closed. You can take over tests from branch ar/filecoinfix .

Please close pr.

**Pull request closed without merging**